design/ceph/object/store.md

```diff
@@ -73,7 +74,83 @@ If `preservePoolsOnDelete` is set to 'true' the pools used to support the object
   preservePoolsOnDelete: true
 ```
 
-If there is a `zone` section in object-store configuration, then the pool section in the ceph-object-zone resource will be used to define the pools.
+#### Pools shared by multiple CephObjectStore
+
+If user want to use existing pools for metadata and data, the pools must be created before the object store is created. This will be useful if multiple objectstore can share same pools. The detail of pools need to shared in `radosNamespaces` settings in object-store CRD. And for this option to work, `preservePoolsOnDelete` must be set to 'true'. Now the object stores can consume same pool isolated with different namespaces. User can create via [Pool CRD](/Documentation/CRDs/Block-Storage/ceph-block-pool-crd.md), this is need to present before the object store is created.
```

Comment:
I think that we should either
1. enforce `preservePoolsOnDelete: true` if namespaces are used, OR
2. ignore the value of `preservePoolsOnDelete` because we know that the pools need preserved, being defined by CephBlockPool CRs.

This is also a good thing to note in the design because it exposes the need for us to have a similar setting for namespaces.

I would offer that we might want to create a new configurable `preserveDataOnDelete` which preserves pools if namespacing isn't used, or preserves the RADOS namespace if namespacing is used. We can then deprecate (but keep around for legacy users) the `preservePoolsOnDelete` config.

@travisn thoughts?

Reply:
Yes, that sounds good to have `preserveDataOnDelete` and deprecate the other setting. If the object store is using RADOS namespaces, we should anyway never delete the pools since they would be owned by the CephBlockPool CRs.

Reply:
@thotz reminder that this has not been addressed yet.

Reply:
Yaa understand the usecase, but I was not able to find a way to list or delete namespaces in Ceph.

Reply:
I have used the `rados` tool to work with RBD namespaces

Reply:
@BlaineEXE there is option `-N` to give namespace, but nothing to list the namespace as such. There is this wiki page bit old https://tracker.ceph.com/projects/ceph/wiki/Rados_namespaces. Man page does not give much options either https://docs.ceph.com/en/latest/man/8/rados/

Reply:
Can please the share docs or commands, and see whether it works for RGW as well like in RBD